services/web-app/utils/seo.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { assetsNavData, globalNavData, standardsNavData } from '@/data/nav-data'
+
+/**
+ * Helper function to conditionally remove the last portion of a path.
+ *
+ * @param {string} path
+ * @param {boolean} popPath
+ * @returns {string}
+ */
+const getPath = (path, popPath) => {
+  if (popPath) {
+    return path?.split('/')?.slice(0, -1)?.join('/')
+  }
+
+  return path
+}
+
```

Comment:
I feel like doing it recursively is a bit easier to read in this case?, totally optional!:
```suggestion

/**
 * For a given side nav path like `/about-carbon/how-carbon-works`,
 *  if that matches any supplied nav data,
 * construct a title like "How Carbon works - About Carbon" to ensure all titles are unique. The
 * lowest node in the tree is shown first in the meta title.
 *
 * @param {Array} navData - array of all nav items to find sideNavPath in
 * @param {string} sideNavPath - path to side nav item to find
 * @param {boolean} hasTabs - if the page has page tabs
 * @returns {string} title
 */
const getPageTitle = (navData, sideNavPath, hasTabs, parentTitle = '') => {
  for (const navItem of navData) {
    const navItemTitle = parentTitle ? `${navItem.title} - ${parentTitle}` : navItem.title
    if (getSideNavPath(navItem.path, hasTabs) === sideNavPath) {
      return navItemTitle
    }
    if (Array.isArray(navItem.items)) {
      const title = getPageTitle(navItem.items, sideNavPath, hasTabs, navItemTitle)
      if (title) {
        return title
      }
    }
  }
}

```